.github/workflows/analyze-modified-files.yml

```
@@ -53,7 +53,7 @@ jobs:
       - uses: actions/setup-python@v5
         if: env.diff != ''
         with:
-          python-version: 3.8
+          python-version: 3.10
```

> I also don't currently know why this fails

This isn't quoted, so it's treated as a YAML float and the YAML parser "helpfully" decides that 3.10 is thus the same as 3.1

```suggestion
          python-version: '3.10'
```

---

That would do it. Thanks for spotting it.